documentation update (#170)
ltelab/disdrodb: docs/source/data.rst

=========================

> Users can make their own data accessible to the community. Disdrodb

Reviewer: ```suggestion
Users can make their own data accessible to the community. DISDRODB
```

provides a central storage for code (readers), issues and metadata.
However, the raw data itself must be stored by the data provider due to
size limitations.

Two types of data must be distinguished:

-  Station Raw Data:

   -  Stores disdrometer measurements for days, weeks, and years.
   -  This dataset can be very heavy.
   -  No central storage is provided.

-  Station Metadata and Issues:

   -  Stores metadata on disdrometer measurements or measurement issues.

>    -  This dataset should be light.

Reviewer: Maybe this could be removed

   -  Central storage is provided in the ``disdro-data`` Git repository.
   -  The metadata folder contains a YAML metadata file called

>       ``metadata.yml``. It has a ``data_url`` key that references the

Reviewer: ```suggestion
      ``metadata.yml``. It has a ``data_url`` key that references to the remote/online repository where
```

>       raw data. Only one zip file per station is allowed.

Reviewer: ```suggestion
      station's raw data are stored. At this URL, a single zip file provides all data available for a given station.

> How to download metadatas and issues ?

Reviewer: ```suggestion
How to download the DISDRODB metadata and issues archive ?
```

--------------------------------------

> You can clone the disdrodb-data repository with

Reviewer: ```suggestion
You can clone the disdrodb-data repository with:
```

.. code:: bash

   git clone https://github.com/ltelab/disdrodb-data.git

However, if you plan to add new data or metadata to the archive, first
fork the repository on your GitHub account and then clone the forked
repository.

> How to upload metadatas and issues?

Reviewer: ```suggestion
How to upload new station metadata?
```

------------------------------------

Do you want to contribute to the project with your own data? Great! Just
follow these steps:

1. Fork the ``disdro-data`` Git repository.

> 2. Create a new branch.

Reviewer: ```suggestion
2. Create a new branch:
```

   .. code:: bash

      git checkout -b "reader-<data_source>-<campaign_name>"

> 3. Add your data source, campaign names, and station name to the current

Reviewer: For metadata you don't have to add a directory <station_name>. Maybe here we should remember to call themetadata YAML file with`<station_name>.yml`

>    folder structure.

Reviewer: ```suggestion
   disdrodb-data structure.
```

> 4. Add your metadata YAML file for each station ``station_name.yml``, in

Reviewer: ```suggestion
4. Add your metadata YAML file for each station following the name format convention ``<station_name>.yml``, in
```

>    a metadata directory in the campaign directory. We recommend you

Reviewer: ```suggestion
   We recommend you
```

   copy-paste an existing metadata YAML file to get the correct
   structure.

5. (Optional) Add your issues YAML files, for each station

>    ``station_name.yml``, in an issue directory located in the campaign

Reviewer: ```suggestion
   ``station_name.yml``, in an  ``issues`` directory located in the campaign
```

>    directory. We recommend you copy-paste an existing metadata YAML file

Reviewer: ```suggestion
   directory. We recommend you to copy-paste an existing issue YAML file
```

   to get the correct structure.

> 6. Test that the integration of your new dataset functions by deleting

Reviewer: Not clear here. You mean after that the PR has been merged, or after that you push on the personal fork?
Here I would add a BIG WARNING: to first git add, commit and push code on the repo ... to avoid loosing the work done !

Reviewer: This should go after point 7. no?

   your data locally and re-fetching it through the process detailed
   above.

7. Commit your changes and push your branch to GitHub.

8. `Create a pull
   request <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request>`__,
   and wait for a maintainer to accept it!

9. If you struggle with this process, don’t hesitate to raise an
   `issue <https://github.com/ltelab/disdrodb-data/issues/new/choose>`__
   so we can help!

How to download raw data locally ?
--------------------------------------

> Prerequisite: You have already set up the folder hierarchy as described

Reviewer: I would rather say. 
1. First clone the disdrodb-data repository.
2. The retrieve the path to disdrodb-data/DISDRODB: this will be the disdrodb_dir root directory !

Here we could also suggest to add the path to the disdrodb config file (if we create this). 
I think would short a lot the command line routines and easy the use
Then

above.

> Objective: You would like to download the raw data referenced in the

Reviewer: ```suggestion
Objective: You would like to download the raw data referenced in some metadata
```

``metadata.yml`` file.

> In order to download data, you should be in the virtual environment.

Reviewer: ```suggestion
In order to download the data, you should be in a virtual environment with the disdrodb package installed !
```

To download all data, just run:

.. code:: bash

>    download_disdrodb_archive  <the_root_folder> --data_sources <data_souzrce> --campaign_names <capaign_name> --station_names <station_name> --force true

Reviewer: ```suggestion
   download_disdrodb_archive  <the_root_folder> --data_sources <data_source> --campaign_names <campaign_name> --station_names <station_name> --force true
```

The ``disdrodb_dir`` parameter is compulsory and must include the path
of the root folder, ending with ``DISDROD``. The other parameters are
optional and are meant to restrict the download processing to a specific
data source, campaign, or station.

Parameters:

> -  ``data_sources`` (optional): the source of the data

Reviewer: Use the classical way of define data_source ... institute/organization ...

-  ``campaign_names`` (optional): the name of the campaign.
-  ``station_names`` (optional): the name of the station.
-  ``force`` (optional, default = ``False``): a boolean value indicating

>    whether existing files should be overwritten.

Reviewer: Here add a line saying that data from multiple data sources (or campaigns) want to be downloaded ... to specify with a space separated string i.e "EPFL NASA"

How to upload raw data to Zenodo?
---------------------------------

We provide users with a code to upload their station’s raw data to
Zenodo.

.. code:: bash

   upload_disdrodb_archive <the_root_folder> --data_sources <data_source> --campaign_names <campaign_name> --station_names <station_name> --platform <name_of_the_platform> --force true

The ``disdrodb_dir`` parameter is compulsory and must include the path

> of the root folder, ending with ``DISDROD``. The other parameters are

Reviewer: ```suggestion
of the root folder, ending with ``DISDRODB``. The other parameters are
```

optional and are meant to restrict the upload processing to a specific
data source, campaign, or station.

Parameters:

> -  ``data_sources`` (optional): the source of the data.

Reviewer: Modify also here ...

-  ``campaign_names`` (optional): the name of the campaign.
-  ``station_names`` (optional): the name of the station.
-  ``platform`` (optional, default is Zenodo).

> -  ``force`` (optional, default = ``False``): a boolean value indicating

Reviewer: Is this really implemented?

   whether files already uploaded somewhere else should still be
   included.

Currently, only Zenodo is supported.

After running this command, the user will be prompted to insert a Zenodo
token. Once the data is uploaded, a link will be displayed that the user
must use to go to the Zenodo web interface and manually publish the
data.

> To get a Zenodo token, go to

Reviewer: Provide hyperlink here:

      ``metadata.yml``. It has a ``data_url`` key that references to the remote/online repository where
      station's raw data are stored. At this URL, a single zip file provides all data available for a given station.

Data transfer upload and download schema :

> .. image:: /static/transfer.png

Reviewer: I have some suggestion to improve the figure (which is already great). 
- Instead of `DisdroDB official repository` --> `DISDRODB metadata archive repository`
- Add background to git command. For example `git clone + pull`
- Maybe the term `update` is more appropriate than `upload` for the metadata archive.
- Maybe you can add also a dashed arrow from `DISDRODB Download/Upload Utility` to `Stations raw data` and `Personal online repository`

> DISDRODB metadata and issues archive download

Reviewer: ```suggestion
Download the DISDRODB metadata archive
```

fork the repository on your GitHub account and then clone the forked
repository.

> DISDRODB metadata and issues archive upload

Reviewer: ```suggestion
Update the DISDRODB metadata archive
```

   `issue <https://github.com/ltelab/disdrodb-data/issues/new/choose>`__
   so we can help!

> DISDRODB raw data download

Reviewer: ```suggestion
Retrieve the DISDRODB raw data archive
```

the data sources or campaigns you require. For example, "EPFL NASA".

> DISDRODB raw data upload to Zenodo

Reviewer: ```suggestion
Add new data to the DISDRODB raw data archive (using Zenodo)
```
